### ISSUE TITLE
Support Typescript components

### ISSUE THREAD

**Issue by nirhart** [wix/wix-style-react#819]
Allow to write part of the components using typescript

**Comment by tqmukas** [comment]
I would like to understand what this feature actually solves. It makes no sense to me to introduce TypeScript types while all the components are currently being protected by `prop-types`. Looks like a little bit of an overhead. I suggest us to find and analyse a good use case before starting any implementation. It will require pretty big effort not only to implement but also to support it in the future.

@lbelinsk @argshook

**Comment by argshook** [comment]
this definitely is not a priority, i wouldn't invest time into this at the moment.

what we care about are props and we cover them with `propTypes`. There are occasional mismatches with them and so aligning those would be a better area to spend effort on.

**Comment by lbelinsk** [comment]
started to do that in wix-ui new lib.
